storage/blob_test.go

```diff
@@ -71,6 +71,63 @@ func (s *StorageBlobSuite) TestGetBlobSASURI(c *chk.C) {
 	c.Assert(expectedParts.Query(), chk.DeepEquals, sasParts.Query())
 }
 
+// Newer version specified but cannot specify IP range since this isn't really a unit test
+// but an intergration test and have no idea what the external IP address is.
+func (s *StorageBlobSuite) TestGetBlobSASURIWithSignedIPAndProtocolValidAPIVersionPassed(c *chk.C) {
+	api, err := NewClient("foo", "YmFy", DefaultBaseURL, "2015-04-05", true)
+	c.Assert(err, chk.IsNil)
+	cli := api.GetBlobService()
+	expiry := time.Time{}
+
+	expectedParts := url.URL{
+		Scheme: "https",
+		Host:   "foo.blob.core.windows.net",
+		Path:   "container/name",
+		RawQuery: url.Values{
+			"sv":  {"2015-04-05"},
+			"sig": {"R1zAMfObctZ+drkn1KJAl+9NOsffWjocV6rFXV/eySI="},
+			"sr":  {"b"},
+			"sp":  {"r"},
+			"se":  {"0001-01-01T00:00:00Z"},
+			"spr": {"https"},
+		}.Encode()}
+
+	u, err := cli.GetBlobSASURIWithSignedIPAndProtocol("container", "name", expiry, "r", "", []string{"https"})
+	c.Assert(err, chk.IsNil)
+	sasParts, err := url.Parse(u)
+	c.Assert(err, chk.IsNil)
+	c.Assert(sasParts.String(), chk.Equals, expectedParts.String())
```

Comment:
My bad, though I did change to to sasParts.Query() and expectedParts.Query() as per pre-existing tests